packages/grid/_modules_/grid/models/gridSlotsComponent.ts

```diff
@@ -51,6 +52,14 @@ export interface GridSlotsComponent extends GridIconSlotsComponent {
    * Header component rendered above the grid column header bar.
```

Comment:
Why do we duplicate the component keys? Shouldn't we have a single source of truth?

```suggestion
   * Prefer using the `Toolbar` slot. You should never need to use this slot. TODO remove.
```

Reply:
As far as I can see the only difference is that here there are optional. Maybe we can just use `Partial` to achieve that @dtassone?

Reply:
It used to map the components with the base props